From: SeongJae Park <sj@kernel.org>
Date: Fri, 4 Jul 2025 11:39:08 -0700
Subject: [PATCH] mm/damon/core: add cleanup_target() ops callback

For example, vaddr use case needs to put pids.  Let it be done in vaddr,
using cleanup_target() callback.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index f98451bd58ab..83741079e19b 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -686,6 +686,7 @@ enum damon_ops_id {
  * filters (&struct damos_filter) that handled by itself.
  * @target_valid should check whether the target is still valid for the
  * monitoring.
+ * @cleanup_target is called before the target will be deallocated.
  * @cleanup is called from @kdamond just before its termination.
  */
 struct damon_operations {
@@ -703,6 +704,7 @@ struct damon_operations {
 			struct damon_target *t, struct damon_region *r,
 			struct damos *scheme, unsigned long *sz_filter_passed);
 	bool (*target_valid)(struct damon_target *t);
+	void (*cleanup_target)(struct damon_target *t);
 	void (*cleanup)(struct damon_ctx *context);
 };
 
-- 
2.39.5

